src/Paket/Program.fs

```diff
@@ -49,6 +49,9 @@ type CLIArguments =
     | [<AltCommandLine("-f")>] Force
     | Hard
     | [<CustomCommandLine("nuget")>] Nuget of string
+//    | [<CustomCommandLine("gist")>] Gist of string * string
+//    | [<CustomCommandLine("github")>] Github of string * string
+    | [<CustomCommandLine("url")>] Url of string * string
     | [<CustomCommandLine("version")>] Version of string
     | [<CustomCommandLine("add-credentials")>] AddCredentials of string
```

Comment:
You might also want to look at https://github.com/fsprojects/Paket/blob/master/src/Paket/Program.fs#L45 and https://github.com/fsprojects/Paket/blob/master/src/Paket/Program.fs#L189

with this trick you don't get string * string option but a list of strings. That might be enough. You could then just throw if you have more than 2 strings